File: src/test/kotlin/fibonacci.kt

fun fibonacci(n: Int): Int {
    if (n == 0) return 0
    if (n == 1) return 1
    return fibonacci(n - 1) + fibonacci(n - 2)
}

fun fibonacci2(n: Int): Int {
    var a = 0
    var b = 1
    var c = 0
    for (i in 1..n) {
        c = a + b
        a = b
        b = c
    }
    return a
}

fun main(args: Array<String>) {
    println(fibonacci(42))
}

File: src/test/polyglot/fibonacci.rb
puts 'Fibonacci in Ruby on GraalVM!'

n = 42

def fibonacci(n)
    if n == 1
        1
    elsif n == 2
        1
    else
        fibonacci(n-1) + fibonacci(n-2)
    end
end

def fibonacci_iterative(n)
    a = 1
    b = 1
    for i in 3..n
        c = a + b
        a = b
        b = c
    end
    b
end

puts "#{n}'s fibonacci value is #{fibonacci(n)}"